src/ripple/basics/impl/make_SSLContext.cpp

```diff
@@ -154,16 +163,28 @@ static void x509_sign (X509* x509, EVP_PKEY* evp_pkey)
         LogicError ("X509_sign failed");
 }
 
-static void ssl_ctx_use_certificate (SSL_CTX* const ctx, x509_ptr& cert)
+static void ssl_ctx_use_certificate (SSL_CTX* const ctx, x509_ptr cert)
 {
-    if (SSL_CTX_use_certificate (ctx, cert.release()) <= 0)
+    if (SSL_CTX_use_certificate (ctx, cert.get()) <= 0)
         LogicError ("SSL_CTX_use_certificate failed");
+    {
+        static std::mutex m;
+        static x509_ptr cleanupMem;
+        std::lock_guard<std::mutex> l{m};
```

Comment:
Note: I threw in this lock at the last minute, but it's probably not needed. I should either get rid of it, or make `cleanupMem` a `std::vector<x509_prt>` so it really could handle multiple instances if needed.